File: JavaScript/FindBottomLeftTreeValue/FindBottomLeftTreeValue.js
/**
 * Created by Phoenix-Z on 2017/4/20.
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * (139ms)
 * @param {TreeNode} root
 * @return {number}
 */
var findBottomLeftValue = function(root) {
    var treeVal = [];
    traverse(root, treeVal, 0);
    return treeVal.pop().shift();
};

var traverse = function (root, treeVal, level) {
    if(root === null)
        return;
    if(level >= treeVal.length) {
        var now = [root.val];
        treeVal.push(now);
    } else {
        treeVal[level].push(root.val);
    }
    traverse(root.left, treeVal, level + 1);
    traverse(root.right, treeVal, level + 1);
};

File: Python/FindBottomLeftTreeValue/FindBottomLeftTreeValue.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
	# (88ms)
    def findBottomLeftValue(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def traverse(root, treeVal, level):
            if not root:
                return
            if level >= len(treeVal):
                treeVal.append([root.val])
            else:
                treeVal[level].append(root.val)
            traverse(root.left, treeVal, level + 1)
            traverse(root.right, treeVal, level + 1)
        treeVal = []
        traverse(root, treeVal, 0)
        return treeVal[-1][0]